components/pingfederate.key.manager/src/main/java/org/wso2/pingfederate/client/PingFederatKeyManagerClient.java

```diff
@@ -176,19 +182,54 @@ private ClientInfo fromOauthAppRequestToClientInfo(OAuthAppRequest oAuthAppReque
             clientInfo.setClientId(UUID.randomUUID().toString());
         }
         if (additionalProperties.containsKey(APIConstants.JSON_CLIENT_SECRET)) {
-            clientInfo.setClientId((String) additionalProperties.get(APIConstants.JSON_CLIENT_SECRET));
+            clientInfo.setSecret((String) additionalProperties.get(APIConstants.JSON_CLIENT_SECRET));
         } else if (StringUtils.isNotEmpty(oAuthApplicationInfo.getClientSecret())) {
             clientInfo.setSecret(oAuthApplicationInfo.getClientSecret());
-        } else {
+        } else if (!isUpdateFlow) {
             clientInfo.setSecret(UUID.randomUUID().toString());
         }
         if (additionalProperties.containsKey(PingFederateConstants.BYPASS_APPROVAL_PAGES)) {
             clientInfo.setBypassApprovalPage(Boolean.parseBoolean(
-                    (String) additionalProperties.get(PingFederateConstants.BYPASS_APPROVAL_PAGES)));
+                    additionalProperties.get(PingFederateConstants.BYPASS_APPROVAL_PAGES).toString()));
         }
         if (additionalProperties.containsKey(PingFederateConstants.RESTRICT_RESPONSE_TYPES)) {
-            clientInfo.setBypassApprovalPage(Boolean.parseBoolean(
-                    (String) additionalProperties.get(PingFederateConstants.BYPASS_APPROVAL_PAGES)));
+            if (additionalProperties.get(PingFederateConstants.RESTRICT_RESPONSE_TYPES) instanceof String &&
+                    StringUtils.isEmpty((String)
+                            additionalProperties.get(PingFederateConstants.RESTRICT_RESPONSE_TYPES))) {
+                clientInfo.setRestrictedResponseTypes(Collections.emptyList());
+            } else {
+                clientInfo.setRestrictedResponseTypes(
+                        (List<String>) additionalProperties.get(PingFederateConstants.RESTRICT_RESPONSE_TYPES));
+            }
+        }
+        if (additionalProperties.containsKey(PingFederateConstants.CLIENT_AUTHENTICATION_TYPE)) {
+            clientInfo.setClientAuthnType(
+                    (String) additionalProperties.get(PingFederateConstants.CLIENT_AUTHENTICATION_TYPE));
```

Comment:
You will have to do a `instanceof String` check before doing a cast or do a null check and use `toString()`

Reply:
Fixed with bbb6420c476442a861796b7469840258c013f066